### PR TITLE
Resolves #6751 - added kafka consumer assign method interception to p…

### LINKED ISSUE #6751: [Feature] support for kafka consumer assign in kafka-plugin
There is currently only support for subscribe method usage in kafka consumer.
Another option in kafka is to assign topic partition using method 'assign' instead of subscribe.
Lack of this is causing traces to display 'null' topics for assigned consumers.
I will provide PR for this.

### PR DESCRIPTION
**Pull request by Yebemeto**
…roperly trace requests going through 'assigned' topics

- [X ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #6751.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
I think you should learn about plugin test
1. https://github.com/apache/skywalking/tree/master/test/plugin/scenarios/kafka-scenario should be updated, including test demo codes and expected data file.
2. Changes.md should be updated.

**Comment by dmsolr** [comment]
@Yebemeto Hi, cloud you like to collect partition numbers when you collect the topics?

**Comment by wu-sheng** [comment]
@Yebemeto Any update?

**Comment by Yebemeto** [comment]
@wu-sheng  Hey, I ll take a look very soon, perhaps today.

**Comment by Yebemeto** [comment]
@dmsolr In use case i need this for it is not beneficial. You think it should be recorded as part of trace?
This would kind of change how both subscribe and assign traces look in kafka.

**Comment by dmsolr** [comment]
> @dmsolr In use case i need this for it is not beneficial. You think it should be recorded as part of trace?
> This would kind of change how both subscribe and assign traces look in kafka.

I got it. Thanks for your explanation.

File: apm-sniffer/apm-sdk-plugin/kafka-plugin/src/test/java/org/apache/skywalking/apm/plugin/kafka/AssignMethodInterceptorTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        @Override
+        public void setSkyWalkingDynamicField(Object value) {
+            this.consumerEnhanceRequiredInfo = (ConsumerEnhanceRequiredInfo) value;
+        }
+    };
+
+    @Before
+    public void setUp() {
```
Comment (darcydai) [inline apm-sniffer/apm-sdk-plugin/kafka-plugin/src/test/java/org/apache/skywalking/apm/plugin/kafka/AssignMethodInterceptorTest.java:58]: setup

**Comment by zifeihan** [comment]
Please check CI status, some checkStyle was failed.

File: apm-sniffer/apm-sdk-plugin/kafka-plugin/src/test/java/org/apache/skywalking/apm/plugin/kafka/AssignMethodInterceptorTest.java (hunk 1 above)
Comment (Yebemeto) [reply apm-sniffer/apm-sdk-plugin/kafka-plugin/src/test/java/org/apache/skywalking/apm/plugin/kafka/AssignMethodInterceptorTest.java]: thanks, done.

**Comment by Yebemeto** [comment]
Ok, thanks for your input everybody. I think i have now done everything.
@wu-sheng I made plugin test cases for this and also updated CHANGES.md
@zifeihan i fixed code style

**Comment by wu-sheng** [comment]
@darcydai @dmsolr @zifeihan Please review.

**Review by zifeihan — APPROVED** [review]
thanks, LGTM

**Review by dmsolr — APPROVED** [review]
lgtm

**Pull request merged**